CMakeLists.txt

```diff
@@ -413,6 +413,20 @@ if (NOT EXTERNAL_MODULES_LOCATION STREQUAL "")
 	endforeach()
 endif()
 
+#=============================================================================
+# Testing - Automatic unit and integration testing with CTest
+#
+
+option(unit_testing "Configure unit test targets" OFF)
```

Comment:
Can the existing cmake testing facilities be used directly?

`enable_testing()` - https://cmake.org/cmake/help/v3.8/command/enable_testing.html?highlight=i
Then you can check `BUILD_TESTING `.https://github.com/PX4/Firmware/blob/d42bb01e0ca83813f1d8a96576c6fb9244bb7a19/platforms/posix/CMakeLists.txt#L117

Reply:
I wanted to do that but they are used already for the SITL integration tests. I'd like to be able to run the tests separately since the SITL tests take so much longer. I planned to have a structured test name and use `ctest -R` to filter the correct ones. So you think distinguishing build time doesn't make sense? I could agree with that. Then gtest is always needed for tests in general and it all depends on `BUILD_TESTING`.

Reply:
Ah and testing is always enabled for sitl... https://github.com/PX4/Firmware/blob/fc7c7ac20630951b17e8644ccc175b9c4e743409/CMakeLists.txt#L303-L305
Can I make it optional?

Reply:
> since the SITL tests take so much longer

Are you talking about the SITL unit tests? Each is roughly 1-3 seconds.
Example: http://ci.px4.io:8080/blue/organizations/jenkins/PX4%2FFirmware/detail/master/722/pipeline/21

![Screen Shot 2019-03-09 at 3 45 14 PM](https://user-images.githubusercontent.com/84712/54077129-62116680-4282-11e9-9e26-6bf13952a92b.png)

Reply:
Yes, I expected the basic tests to run two order of magnitudes faster. Is that impatient? 😇 I think it's just starting the simulator binary that takes the big part of that time, otherwise the smallest tests would not all have around the same duration.
Anyways I think we should allow to run the unit tests separately since I presume they'll finish a lot faster until we write a crazy amount of tests or huge loops.

Reply:
I'll just add all tests and the gtest library fetching with `enable_testing()` and hence `BUILD_TESTING` but make enabling tests optional for posix. Then I can still filter the tests if I only want to run the quick unit tests e.g. with `make tests unit`.